waspc/src/Wasp/Generator/WebAppGenerator.hs

```diff
@@ -320,10 +320,13 @@ genViteConfig spec = return $ C.mkTmplFdWithData tmplFile tmplData
           "defaultClientPort" .= C.defaultClientPort
         ]
 
-    makeCustomViteConfigJsImport :: Path' (Rel SourceExternalCodeDir) File' -> JsImport
+    makeCustomViteConfigJsImport :: Path' (Rel WaspProjectDir) File' -> JsImport
     makeCustomViteConfigJsImport pathToConfig = makeJsImport (RelativeImportPath importPath) importName
       where
-        importPath = C.toViteImportPath $ fromJust $ SP.relFileToPosix pathToConfigInSrc
-        pathToConfigInSrc = SP.castRel $ C.webAppSrcDirInWebAppRootDir </> SP.castRel pathToConfig
+        importPath = SP.castRel $ C.toViteImportPath relPathToConfigInProjectDir
+        relPathToConfigInProjectDir = relPathFromWebAppRootDirWaspProjectDir </> (fromJust . SP.relFileToPosix $ pathToConfig)
+
+        relPathFromWebAppRootDirWaspProjectDir :: Path Posix (Rel C.WebAppRootDir) (Dir WaspProjectDir)
+        relPathFromWebAppRootDirWaspProjectDir = [reldirP|../../..|]
```

Comment:
We already have this knowledge in our codebase. You can get what you need by combining `webAppRootDirInProjectRootDir` (in WebAppGenerator.Common) and `webAppRootDirInProjectRootDir` (in Project.Common).

Functions you'll need:
- `relDirToPosix`: https://hackage.haskell.org/package/strong-path-1.1.4.0/docs/StrongPath.html#g:26
- `reversePosixPath` in `FilePath.Extra`

Reply:
Nice stuff, I'll do it like this 👍